File: Aula_Ricardo/Programação_orientada_a_objetos/go/Agregação.go
package main

import "fmt"

type Funcionario struct {
	Nome    string
	Cargo   string
	Salario float64
}

func (f Funcionario) String() string {
	return fmt.Sprintf("Funcionario Nome: %s, Cargo: %s, Salario: %.2f", f.Nome, f.Cargo, f.Salario)
}

type Empresa struct {
	Funcionarios []Funcionario
}

func (e *Empresa) AdicionarFuncionario(funcionario Funcionario) {
	e.Funcionarios = append(e.Funcionarios, funcionario)
}

func (e Empresa) ListarFuncionarios() {
	for _, funcionario := range e.Funcionarios {
		fmt.Println(funcionario)
	}
}

func main() {

	empresa := Empresa{}

	func1 := Funcionario{Nome: "Pedro", Cargo: "Vigia", Salario: 2000}

	empresa.AdicionarFuncionario(func1)

	empresa.ListarFuncionarios()
}

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Agregação.java
import java.util.ArrayList;
import java.util.List;

class Funcionario {
    private String nome;
    private String cargo;
    private double salario;

    public Funcionario(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    @Override
    public String toString() {
        return "Funcionario Nome: " + nome + ", Cargo: " + cargo + ", Salario: " + salario;
    }
}

class Empresa {
    private List<Funcionario> funcionarios;

    public Empresa() {
        funcionarios = new ArrayList<>();
    }

    public void adicionarFuncionario(Funcionario funcionario) {
        funcionarios.add(funcionario);
    }

    public void listarFuncionarios() {
        for (Funcionario funcionario : funcionarios) {
            System.out.println(funcionario);
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Empresa empresa = new Empresa();

        Funcionario func1 = new Funcionario("Pedro", "Vigia", 2000);

        empresa.adicionarFuncionario(func1);

        empresa.listarFuncionarios();
    }
}
